src/components/calcite-switch/calcite-switch.tsx

```diff
@@ -21,34 +22,107 @@ import { Scale, Theme } from "../interfaces";
   shadow: true
 })
 export class CalciteSwitch {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
   @Element() el: HTMLCalciteSwitchElement;
 
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
   /** True if the switch is disabled */
   @Prop({ reflect: true }) disabled?: boolean = false;
 
-  /** The name of the checkbox input */
-  @Prop({ reflect: true, mutable: true }) name?: string = "";
+  @Watch("disabled")
+  disabledWatcher(newDisabled: boolean): void {
+    this.inputEl.disabled = newDisabled;
+  }
+
+  /** The id attribute of the switch.  When omitted, a globally unique identifier is used. */
+  @Prop({ reflect: true, mutable: true }) guid: string;
+
+  /** The name of the switch input */
+  @Prop({ reflect: true }) name?: string = "";
```

Comment:
Does the name need to be an empty string by default? Just thinking that the input shouldn't have a name attribute unless its defined.